torchdynamo/convert_frame.py

```diff
@@ -96,6 +97,28 @@ def _fn(*args, **kwargs):
     return _fn
 
 
+def has_tensor_in_frame(frame):
+    # Check if the passed arguments are of type Tensor
+    for _, value in frame.f_locals.items():
+        if value is not None and isinstance(value, torch.Tensor):
+            return True
+
+    # Check if there is global import of torch.*
+    allowed_modules = ["torch"]
+    for _, value in frame.f_globals.items():
```

Comment:
`frame.f_code.co_names` contains a list of all the global variable names this function could read.  

So we don't actually need to iterate over all globals, just the ones that the code could reference.